streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -258,6 +260,8 @@ private boolean hasRestoredToEnd(final ChangelogMetadata metadata) {
                 // if we cannot get the position of the consumer within timeout, just return false
                 return false;
             } catch (final KafkaException e) {
+                // this also includes InvalidOffsetException, which should not happen under normal
```

Comment:
Just wondering, why is it ok to get `InvalidOffsetException` during `restore`/`poll` but not here? When might this get thrown from `#position`? ditto for in `prepareChangelogs` down below

Reply:
`consumer.poll` throwing `InvalidOffsetException` should be handled as TaskCorrupted; `consumer.position` throwing `InvalidOffsetException` should not happen under normal scenarios, when it happens it indicates a bug and hence we do not need to special handle it.